### ISSUE TITLE
Sepolia CL clients report error, Erigon is on wrong head

### ISSUE THREAD

**Issue by barnabasbusa** [ledgerwatch/erigon#4866]
#### System information

Erigon version: `erigon version 2022.07.4-alpha-d88fc824`

OS & Version: Ubuntu 22.04

Commit hash : d88fc8242

#### Expected behaviour
`eth_blockNumber` returning the last valid block. 

#### Actual behaviour
`eth_blockNumber` returns a different block.

#### Steps to reproduce the behaviour
Try to sync up sepolia testnet with erigon + teku or erigon + nimbus (it is possible that other clients also don't work, these are just the two I have tested so far). 

API call `eth_syncing` returns the following:

```
Geth + teku:
{"jsonrpc":"2.0","id":1,"result":false}

Erigon + nimbus:
{"jsonrpc":"2.0","id":1,"result":false}

Besu + prysm:
{
  "jsonrpc" : "2.0",
  "id" : 1,
  "result" : false
}
Nethermind + lighthouse:
{"jsonrpc":"2.0","result":false,"id":1}
Geth2 + lodestar :
{"jsonrpc":"2.0","id":1,"result":false}

Erigon2 + teku:
{"jsonrpc":"2.0","id":1,"result":false}
```
Therefore, I assume that the chain is fully synced, however when I run the API call `eth_blockNumber` I get the following

```
Geth + teku:
{"jsonrpc":"2.0","id":1,"result":"0x18687f"}

Erigon + nimbus:
{"jsonrpc":"2.0","id":1,"result":"0x184e2c"}

Besu + prysm:
{
  "jsonrpc" : "2.0",
  "id" : 1,
  "result" : "0x18687f"
}
Nethermind + lighthouse:
{"jsonrpc":"2.0","result":"0x18687f","id":1}
Geth2 + lodestar:
{"jsonrpc":"2.0","id":1,"result":"0x18687f"}

Erigon2 + teku:
{"jsonrpc":"2.0","id":1,"result":"0x17f157"}
```
The correct block is indeed `0x18687f` and not `0x184e2c` or `0x17f157` that are produced by the clients with erigon EL clients. 

Also teku reports an error when connected to erigon:

```
ERROR - PLEASE FIX OR REPORT | Unexpected exception thrown for forkchoice-async-0 java.util.concurrent.CompletionException: java.lang.IllegalStateException: Cannot change node validity from VALID to INVALID (See log file for full stack trace)
```

Similarly the API endpoints of the CLs report inconsistent finalized roots:

```
Teku + geth
{"execution_optimistic":false,"data":{"root":"0xe47f5122d6bc2e84919eac99dbf3cc2bddf9ad9b42ec30f1faa1c9b0c2416f03"}}
Nimbus + erigon
{"execution_optimistic":false,"data":{"root":"0x0997b6522ec31a0c22acb28f9e5c136e4666ca77807a9310e18908443ce2383a"}}
Prysm + besu
{"data":{"root":"0xe47f5122d6bc2e84919eac99dbf3cc2bddf9ad9b42ec30f1faa1c9b0c2416f03"},"execution_optimistic":false}
Lighthouse + nethermind
{"data":{"root":"0xe47f5122d6bc2e84919eac99dbf3cc2bddf9ad9b42ec30f1faa1c9b0c2416f03"}}
Lodestar + geth2
{"data":"0xe47f5122d6bc2e84919eac99dbf3cc2bddf9ad9b42ec30f1faa1c9b0c2416f03"}
Teku2 + erigon2
{"execution_optimistic":false,"data":{"root":"0x1ec1c2fc2aac407e0dffd9fef29fcea486d0d041d07e5e57eaa643505906223a"}}
Nimbus2 + geth3
{"execution_optimistic":false,"data":{"root":"0xe47f5122d6bc2e84919eac99dbf3cc2bddf9ad9b42ec30f1faa1c9b0c2416f03"}}
```

The only common denominator between them is erigon. So I assume there might be something broken in the latest release.

**Comment by yperbasis** [comment]
Might be fixed by PR #4857.

**Comment by barnabasbusa** [comment]
3a972cc0 still hasn't solved it. 

```sh
Teku + geth
{"execution_optimistic":false,"data":{"root":"0xf44b88844ed396756003680f4a73fc7c8ba29da9135027835d72315d58d78ea5"}}
Nimbus + erigon
{"execution_optimistic":false,"data":{"root":"0x0997b6522ec31a0c22acb28f9e5c136e4666ca77807a9310e18908443ce2383a"}}
Prysm + besu
{"data":{"root":"0xf44b88844ed396756003680f4a73fc7c8ba29da9135027835d72315d58d78ea5"},"execution_optimistic":false}
Lighthouse + nethermind
{"data":{"root":"0xf44b88844ed396756003680f4a73fc7c8ba29da9135027835d72315d58d78ea5"}}
Lodestar + geth2
{"data":"0xf44b88844ed396756003680f4a73fc7c8ba29da9135027835d72315d58d78ea5"}
Teku2 + erigon2
{"execution_optimistic":false,"data":{"root":"0xf769d728ab1c90e62c13476d8efa73dcf3ecabceb27e8ea5798c6a80c3e3659c"}}
Nimbus2 + geth3
{"execution_optimistic":true,"data":{"root":"0xf44b88844ed396756003680f4a73fc7c8ba29da9135027835d72315d58d78ea5"}}
```

**Comment by yperbasis** [comment]
Should be fixed by PR #4944; please re-open if it still happens.
